src/audio/module_adapter/module/cadence.c

```diff
@@ -35,6 +35,24 @@ enum cadence_api_id {
 
 #define DEFAULT_CODEC_ID CADENCE_CODEC_WRAPPER_ID
 
+/* we divide Cadence param id space as follows:
+ * - 0x00000000 - 0x7FFFFFFF, Cadence library ids which come
+ *   from xa_*_dec_api.h
+ * - 0x80000000 - 0xFFFFFFFF, SOF remapping of Cadence library ids
```

Comment:
Remapping is required because there is no way we can transmit both AAC and MP3 parameters in the same param id space in the first place as they overlap. More than that they have different IDs.

For example, in order to set pcm width size we need to sent to Cadence library following parameters

MP3:

param_id:   XA_MP3DEC_CONFIG_PARAM_PCM_WDSZ  (= 0),  value = 32

AAC:

param_id:   XA_AACDEC_CONFIG_PARAM_PCM_WDSZ  (= 7), value = 32 

See: 

https://github.com/thesofproject/sof/blob/main/src/include/sof/audio/cadence/aac_dec/xa_aac_dec_api.h#L29

vs

https://github.com/thesofproject/sof/blob/main/src/include/sof/audio/cadence/mp3_dec/xa_mp3_dec_api.h#L32

Reply:
not following this at all.

Is this saying that the parameters in topology need to be formatted so as to match the format expected by a specific implementation? The whole point was to try and push this sort of knowledge in the firmware, userspace and kernel shouldn't depend on what the library requires.